pkg/cluster/formatter.go

```diff
@@ -0,0 +1,121 @@
+package cluster
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/apcera/termtables"
+	"github.com/weaveworks/footloose/pkg/config"
+)
+
+// Formatter formats a slice of machines and outputs the result
+// in a given format.
+type Formatter interface {
+	Format([]*Machine) error
+}
+
+// JSONFormatter formats a slice of machines into a JSON and
+// outputs it to stdout.
+type JSONFormatter struct{}
+
+// NormalFormatter formats a slice of machines into a colored
+// table like output and prints that to stdout.
+type NormalFormatter struct{}
+
+type port struct {
+	Guest int `json:"guest"`
+	Host  int `json:"host"`
+}
+
+type status struct {
+	Name     string         `json:"name"`
+	Spec     config.Machine `json:"spec"`
+	Status   string         `json:"status"`
+	Ports    []port         `json:"ports"`
+	Hostname string         `json:"hostname"`
+}
+
+// Format will output to stdout in JSON format.
+func (JSONFormatter) Format(machines []*Machine) error {
+	statuses := make([]status, 0)
+	for _, m := range machines {
+		s := status{}
+		s.Hostname = m.Hostname()
+		s.Name = m.ContainerName()
+		s.Spec = *m.spec
+		state := "Stopped"
+		if m.IsRunning() {
+			state = "Running"
+		}
+		s.Status = state
+		var ports []port
+		for k, v := range m.ports {
+			p := port{
+				Host:  v,
+				Guest: k,
+			}
+			ports = append(ports, p)
+		}
+		s.Ports = ports
+		statuses = append(statuses, s)
+	}
+	m := struct {
+		Machines []status `json:"machines"`
+	}{
+		Machines: statuses,
+	}
+	ms, err := json.Marshal(m)
```

Comment:
Sure. I just thought that usually when you want JSON you want it because you would like to parse it. In which case indentation might cause a problem in certain cases.

Reply:
json parsers should really be able to handle the whitespace introduced by the indentation :)